core/rawdb/accessors_rollup_event.go

```diff
@@ -143,30 +160,59 @@ func ReadLastFinalizedBatchIndex(db ethdb.Reader) *uint64 {
 
 // WriteCommittedBatchMeta stores the CommittedBatchMeta for a specific batch in the database.
 func WriteCommittedBatchMeta(db ethdb.KeyValueWriter, batchIndex uint64, committedBatchMeta *CommittedBatchMeta) {
-	value, err := rlp.EncodeToBytes(committedBatchMeta)
+	var committedBatchMetaToStore any
+	if encoding.CodecVersion(committedBatchMeta.Version) < encoding.CodecV7 {
+		committedBatchMetaToStore = &committedBatchMetaV0{
+			Version:          committedBatchMeta.Version,
+			ChunkBlockRanges: committedBatchMeta.ChunkBlockRanges,
+		}
+	} else {
+		committedBatchMetaToStore = &committedBatchMetaV7{
+			Version:                committedBatchMeta.Version,
+			ChunkBlockRanges:       committedBatchMeta.ChunkBlockRanges,
+			LastL1MessageQueueHash: committedBatchMeta.LastL1MessageQueueHash,
+		}
+	}
+
+	value, err := rlp.EncodeToBytes(committedBatchMetaToStore)
 	if err != nil {
-		log.Crit("failed to RLP encode committed batch metadata", "batch index", batchIndex, "committed batch meta", committedBatchMeta, "err", err)
+		log.Crit("failed to RLP encode committed batch metadata", "batch index", batchIndex, "committed batch meta", committedBatchMetaToStore, "err", err)
 	}
 	if err := db.Put(committedBatchMetaKey(batchIndex), value); err != nil {
 		log.Crit("failed to store committed batch metadata", "batch index", batchIndex, "value", value, "err", err)
 	}
 }
 
 // ReadCommittedBatchMeta fetches the CommittedBatchMeta for a specific batch from the database.
-func ReadCommittedBatchMeta(db ethdb.Reader, batchIndex uint64) *CommittedBatchMeta {
+func ReadCommittedBatchMeta(db ethdb.Reader, batchIndex uint64) (*CommittedBatchMeta, error) {
 	data, err := db.Get(committedBatchMetaKey(batchIndex))
 	if err != nil && isNotFoundErr(err) {
-		return nil
+		return nil, nil
 	}
 	if err != nil {
-		log.Crit("failed to read committed batch metadata from database", "batch index", batchIndex, "err", err)
+		return nil, fmt.Errorf("failed to read committed batch metadata from database: batch index %d, err: %w", batchIndex, err)
 	}
 
-	cbm := new(CommittedBatchMeta)
-	if err := rlp.Decode(bytes.NewReader(data), cbm); err != nil {
-		log.Crit("Invalid CommittedBatchMeta RLP", "batch index", batchIndex, "data", data, "err", err)
+	// Try decoding from the newest format for future proofness, then the older one for old data.
+	cbm7 := new(committedBatchMetaV7)
+	if err = rlp.Decode(bytes.NewReader(data), cbm7); err == nil {
+		return &CommittedBatchMeta{
+			Version:                cbm7.Version,
```

Comment:
Should we check the version's sanity here? no need if `err = rlp.Decode(bytes.NewReader(data), cbm7); err == nil` can ensure the batch meta is `>= v7`.

Reply:
added in d4cc8978b43920e7e5da43cc494dc59a222c8947